lib/cart.rb

```diff
@@ -0,0 +1,59 @@
+require_relative "menu_item"
+require_relative "menu"
+require_relative "order_item"
+require_relative "customer"
+
+class Cart
+  attr_accessor :cart
+  attr_accessor :total_cost
+
+  def initialize
+    @cart = []
+    @total_cost = 0
+  end
+
+  @@instance = Cart.new
+
+  def self.instance
+    return @@instance
+  end
+
+  def add_item_to_cart(order_item)
+    @cart.push(order_item)
+    cost = order_item.cost.to_i
+    @total_cost += cost
+  end
+
+  def to_s
```

Comment:
So, the way I see this, we are building units. the user interface is a different thing and i think we should keep them totally separated. I would not have any methods use puts or gets, etc. Also I would not have any methods return puts but rather a string if necessary.